### PR TITLE
Update differential module interface, add pre-contrast subsetting options

### PR DESCRIPTION
**Pull request by pinin4fjords**
Closes https://github.com/nf-core/differentialabundance/issues/97

In this PR:

- Update to latest versions of differential modules with newer contrast interface (module guidelines now require value channels for mandatory non-file parameters). 
 - Wire in the updated differential modules' pre-contrast subsetting options in fulfilment of https://github.com/nf-core/differentialabundance/issues/97

## PR checklist

- [x] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the pipeline conventions in the [contribution docs](https://github.com/nf-core/differentialabundance/tree/master/.github/CONTRIBUTING.md)- [ ] If necessary, also make a PR on the nf-core/differentialabundance _branch_ on the [nf-core/test-datasets](https://github.com/nf-core/test-datasets) repository.
- [ ] Make sure your code lints (`nf-core lint`).
- [ ] Ensure the test suite passes (`nextflow run . -profile test,docker --outdir <OUTDIR>`).
- [x] Usage Documentation in `docs/usage.md` is updated.
- [ ] Output Documentation in `docs/output.md` is updated.
- [x] `CHANGELOG.md` is updated.
- [ ] `README.md` is updated (including new tool citations and authors/contributors).

### CODE REVIEW

File: modules/nf-core/limma/differential/templates/limma_de.R [hunk 1]
```diff
@@ -182,7 +185,7 @@ if (length(missing_samples) > 0) {
     ))
 } else{
     # Save any non-count data, will gene metadata etc we might need later
```
Comment (WackerO) [inline modules/nf-core/limma/differential/templates/limma_de.R:187]:
```suggestion
    # Save any non-count data with gene metadata etc we might need later
```

File: modules/nf-core/limma/differential/templates/limma_de.R [hunk 2]
```diff
@@ -224,6 +227,32 @@ if (!contrast_variable %in% colnames(sample.sheet)) {
...
+
+# Optionally, remove samples with specified values in a given field (probably
+# don't use this as well as the above)
+
+if ((! is.null(opt\$exclude_samples_col)) && (! is.null(opt\$exclude_samples_values))){
+    exclude_values = unlist(strsplit(opt\$exclude_samples_values, split = ';'))
+
+    if (! opt\$exclude_samples_col %in% colnames(sample.sheet)){
+        stop(paste(opt\$exclude_samples_col, ' specified to subset samples is not a valid sample sheet column'))
+    }
```
Comment (WackerO) [inline modules/nf-core/limma/differential/templates/limma_de.R:247]:
```suggestion
    if (! opt\$exclude_samples_col %in% colnames(sample.sheet)){
        stop(paste(opt\$exclude_samples_col, ' specified to subset samples is not a valid sample sheet column'))
    }

    exclude_values = unlist(strsplit(opt\$exclude_samples_values, split = ';'))
```

**Review by WackerO — CHANGES_REQUESTED** [review]
Just some details

**Comment by WackerO** [comment]
yes of course, forgot that, it's just for prettying up the code anyway

**Review by WackerO — APPROVED** [review]

**Comment by pinin4fjords** [comment]
Thanks @WackerO !

**Pull request merged**